### PR TITLE
r/glue_data_catalog_encryption_settings - new resource

### LINKED ISSUE #9736: Support for setting AWS Glue Data Catalog Encryption/settings
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

Does not look like there is presently any resource to configure the AWS Glue Data Catalog encryption settings:

* https://docs.aws.amazon.com/glue/latest/dg/encrypt-glue-data-catalog.html
### Potential Terraform Configuration

I would have expected to see something like:

```hcl
resource "aws_glue_data_catalog_settings" "settings" {
    metadata_encryption_kms_key_id = "..."
    password_encryption_kms_key_id = "..."
}
```

... and probably the other bits which are exposed to configure the data catalog settings too?

### PR DESCRIPTION
**Pull request by DrFaust92**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Closes #9736

Release note for [CHANGELOG](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md):

```release-note
resource_aws_glue_data_catalog_encryption_settings - add new resource
```

Output from acceptance testing:

```
$ make testacc TESTARGS='-run=TestAccXXX'
--- PASS: TestAccAWSDataCatalogEncryptionSettings_basic (129.39s)
```

### CODE REVIEW

File: aws/resource_aws_glue_data_catalog_encryption_settings.go [hunk 1]
```diff
@@ -0,0 +1,188 @@
...
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"connection_password_encryption": {
+							Type:     schema.TypeList,
+							Required: true,
+							MaxItems: 1,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"aws_kms_id": {
```
Comment (ewbankkit) [inline aws/resource_aws_glue_data_catalog_encryption_settings.go:40]: Change to `aws_kms_key_id` to match API.

**Comment by ewbankkit** [comment]
Verified acceptance tests:

```console
$ make testacc TEST=./aws TESTARGS='-run=TestAccAWSDataCatalogEncryptionSettings_'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -count 1 -parallel 20 -run=TestAccAWSDataCatalogEncryptionSettings_ -timeout 120m
=== RUN   TestAccAWSDataCatalogEncryptionSettings_basic
=== PAUSE TestAccAWSDataCatalogEncryptionSettings_basic
=== CONT  TestAccAWSDataCatalogEncryptionSettings_basic
--- PASS: TestAccAWSDataCatalogEncryptionSettings_basic (44.63s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	44.672s
```

File: aws/resource_aws_glue_data_catalog_encryption_settings.go (hunk 1 above)
Comment (DrFaust92) [reply aws/resource_aws_glue_data_catalog_encryption_settings.go:40]: fixed

**Comment by ewbankkit** [comment]
@DrFaust92 Acceptance test failure:

```console
$ make testacc TEST=./aws TESTARGS='-run=TestAccAWSDataCatalogEncryptionSettings_'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -count 1 -parallel 20 -run=TestAccAWSDataCatalogEncryptionSettings_ -timeout 120m
=== RUN   TestAccAWSDataCatalogEncryptionSettings_basic
=== PAUSE TestAccAWSDataCatalogEncryptionSettings_basic
=== CONT  TestAccAWSDataCatalogEncryptionSettings_basic
    resource_aws_glue_data_catalog_encryption_settings_test.go:21: Check 5/8 error: aws_glue_data_catalog_encryption_settings.test: Attribute 'data_catalog_encryption_settings.0.connection_password_encryption.0.aws_kms_id' not found
--- FAIL: TestAccAWSDataCatalogEncryptionSettings_basic (7.86s)
FAIL
FAIL	github.com/terraform-providers/terraform-provider-aws/aws	7.905s
FAIL
GNUmakefile:26: recipe for target 'testacc' failed
make: *** [testacc] Error 1
```

**Comment by DrFaust92** [comment]
missed replaces in tests, sorry.

File: aws/resource_aws_glue_data_catalog_encryption_settings_test.go [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+    }
+  ]
+}
+POLICY
+}
+
+resource "aws_glue_data_catalog_encryption_settings" "test" {
+  data_catalog_encryption_settings {
+    connection_password_encryption {
+      aws_kms_id                           = aws_kms_key.test.arn
```
Comment (ewbankkit) [inline aws/resource_aws_glue_data_catalog_encryption_settings_test.go:126]:
```suggestion
      aws_kms_key_id                       = aws_kms_key.test.arn
```

**Comment by DrFaust92** [comment]
`--- PASS: TestAccAWSDataCatalogEncryptionSettings_basic (103.66s)`

**Review by ewbankkit — COMMENTED** [review]
LGTM.

```console
$ make testacc TEST=./aws TESTARGS='-run=TestAccAWSDataCatalogEncryptionSettings_'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -count 1 -parallel 20 -run=TestAccAWSDataCatalogEncryptionSettings_ -timeout 120m
=== RUN   TestAccAWSDataCatalogEncryptionSettings_basic
=== PAUSE TestAccAWSDataCatalogEncryptionSettings_basic
=== CONT  TestAccAWSDataCatalogEncryptionSettings_basic
--- PASS: TestAccAWSDataCatalogEncryptionSettings_basic (45.11s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	45.158s
```

**Comment by breathingdust** [comment]
LGTM! 🚀  Thanks @DrFaust92 

Verified Acceptance Tests in Commercial (us-west-2)

```console
make testacc TEST=./aws TESTARGS='-run=TestAccAWSDataCatalogEncryptionSettings_'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -count 1 -parallel 20 -run=TestAccAWSDataCatalogEncryptionSettings_ -timeout 120m
=== RUN   TestAccAWSDataCatalogEncryptionSettings_basic
=== PAUSE TestAccAWSDataCatalogEncryptionSettings_basic
=== CONT  TestAccAWSDataCatalogEncryptionSettings_basic
--- PASS: TestAccAWSDataCatalogEncryptionSettings_basic (38.55s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	40.551s
```

Verified Acceptance Tests in GovCloud (us-gov-west-1)

```console
make testacc TEST=./aws TESTARGS='-run=TestAccAWSDataCatalogEncryptionSettings_'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -count 1 -parallel 20 -run=TestAccAWSDataCatalogEncryptionSettings_ -timeout 120m
=== RUN   TestAccAWSDataCatalogEncryptionSettings_basic
=== PAUSE TestAccAWSDataCatalogEncryptionSettings_basic
=== CONT  TestAccAWSDataCatalogEncryptionSettings_basic
--- PASS: TestAccAWSDataCatalogEncryptionSettings_basic (28.52s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	30.332s
```

**Pull request merged**